pkg/agent/plugin/workloadattestor/k8s/k8s.go

```diff
@@ -102,23 +107,47 @@ func (p *k8sPlugin) Attest(req *workloadattestor.AttestRequest) (*workloadattest
 	}
 
 	for _, item := range podInfo.Items {
-		for _, status := range item.Status.ContainerStatuses {
-			containerURL, err := url.Parse(status.ContainerID)
-			if err != nil {
-				return &resp, err
-			}
+		match, err := statusMatches(containerID, item.Status)
+		if err != nil {
+			return &resp, err
+		}
 
-			if containerID == containerURL.Host {
-				resp.Selectors = append(resp.Selectors, &common.Selector{Type: selectorType, Value: fmt.Sprintf("sa:%v", item.Spec.ServiceAccountName)})
-				resp.Selectors = append(resp.Selectors, &common.Selector{Type: selectorType, Value: fmt.Sprintf("ns:%v", item.Metadata.Namespace)})
-				return &resp, nil
-			}
+		if match {
+			resp.Selectors = append(resp.Selectors, &common.Selector{Type: selectorType, Value: fmt.Sprintf("sa:%v", item.Spec.ServiceAccountName)})
+			resp.Selectors = append(resp.Selectors, &common.Selector{Type: selectorType, Value: fmt.Sprintf("ns:%v", item.Metadata.Namespace)})
+			return &resp, nil
 		}
 	}
 
 	return &resp, fmt.Errorf("no selectors found")
 }
 
+func statusMatches(containerID string, status podStatus) (bool, error) {
+	for _, status := range status.InitContainerStatuses {
```

Comment:
I'm not sure if we should consider to go over "regular" containers first?
Just bringing this up since (presumably) that would be a more common case and we would find the selectors in the first round. But I don't actually know which would me more commonly used, just bringing this up for consideration.

Reply:
Heh, I have no idea... I had put it first because I thought 1) it comes first in the kubelet response, and 2) there aren't likely to be too many of them. I was wondering actually about performance of this plugin as a whole... We have instrumentation now on a per-attestor basis, so we can start to gather some performance data on these things and see how they're impacting attestation latency.

In the meantime, I have switched them around